lib/CMakeLists.txt

```diff
@@ -1,3 +1,5 @@
-add_library(disccord SHARED client.cpp)
+set(SOURCE_FILES client.cpp)
+
+add_library(disccord SHARED ${SOURCE_FILES})
```

Comment:
It shouldn't be too hard to add an option to build disccord as a static lib using an option. Chances are that's probably what I'll do after jiggling the CMakes about myself (after this PR goes through of course) - using CMake to manage tests as a separate "project" would be easier, I'm hoping.